### ISSUE TITLE
Do you use an external editor for WordPress?

### ISSUE THREAD

**Issue by scripting** [scripting/Scripting-News#263]
A short while ago I had something of a breakthrough -- I thought the Metaweblog API implementation for WordPress was gone, but this was a mistake, it was totally functional and documented. So I put it on my todo list to hook my static site generator, Old School, up to WordPress, so I could use my outliner to write posts for a WordPress blog.

It works. In fact I have it mirroring every post I make on my blog, and have even been able to tune up the CSS so it all looks reasonable. 

Okay so now we know that it can be done, what's the state of external editing for WordPress sites? I imagined that there was a full complement of tools that could work with WordPress, but when I did a quick search I didn't find much. Not sure I'm looking in the right places. So I thought I'd ask readers of my blog what they use, do you all use the internal editors, Calypso and Gutenberg, or do you use an external editor at connects via MetaWeblog?

**Comment by troutcolor** [comment]
Yes, not as often as I used to. But I use TextMate to post via the metaweblog API.  I like it because I have a local copy. Mostly for longer posts. I also tend to post drafts now which lets me tweek some things. 
I use micro.blog to which posts to WordPress and occasionally Mars Edit the Mac desktop app.

**Comment by scripting** [comment]
@troutcolor -- interesting. I'm having a private discussion with a user, I hope he shows up here. 

Here's a link to TextMate. 

https://macromates.com/

Looks like it's positioned as a program editor?

Mars Edit is more of a user-oriented text editor, if I recall correctly.

https://redsweater.com/marsedit/

It looks like both of these are Mac apps. I wonder if there are any web apps that work here?

**Comment by troutcolor** [comment]
TextMate is a programming editor, but I find it one of the most useful apps on my Mac. I turn off lots of “bundles”. The blogging bundle is quite old now and not been updated in an age. 

Mars Edit is just for blogging. It has been regularly updated. 

I also used Fargo to post to my blog for a while but in 2013. 

I used to use a Firefox extension that did metaweblog API. 
The indieWeb folk have MicroPub plug-ins for WordPress that lets you post via some web apps, [Quill](https://quill.p3k.io/) for example but you need to use IndieAuth for identity via another plug-in. 

AppleScript still has XML-RPC support so you can post with that too (just for completion).

**Comment by scripting** [comment]
>>AppleScript still has XML-RPC support

that's surprising.

**Comment by smnevans** [comment]
I often use Ulysses, a Mac-based writing program.

On the web I just use Markdown in the Classic editor. 
I've found Gutenberg to be frustratingly cumbersome for regular text posts.

**Comment by jjn1** [comment]
I write everything in Markdown in Ulysses, which has an Export function to my Wordpress blog. Works seamlessly ( for me, anyway).

**Comment by scripting** [comment]
@jjn1 -- i bet they're using the metaweblog api, which is like the ancient aqueducts all over europe. i was really surprised to find they're still supported by today's wordpress.
